packages/web_ui/src/components/SlavesPage.jsx

```diff
@@ -18,12 +19,24 @@ function GenerateSlaveTokenButton(props) {
 	let control = useContext(ControlContext);
 	let [visible, setVisible] = useState(false);
 	let [token, setToken] = useState(null);
+	let [slaveId, setSlaveId] = useState(null);
 	let [form] = Form.useForm();
 	let tokenTextAreaRef = useRef(null);
+	let [pluginList, setPluginList] = useState([]);
+	useEffect(() => {
+		(async () => {
+			let response = await fetch(`${webRoot}api/plugins`);
+			if (response.ok) {
+				const plugins = await response.json();
+				setPluginList(plugins);
+			} else {
+				notify("Failed to load plugin list");
+			}
+		})();
+	}, []);
 
 	async function generateToken() {
 		let values = form.getFieldsValue();
-		let slaveId = null;
 		if (values.slaveId) {
 			slaveId = Number.parseInt(values.slaveId, 10);
```

Comment:
`slaveId` is changed to a react state variable but still being set directly here, this leads to ill formed behaviour: inputting "a" into the text field then removing it with backspace causes a generateSlaveToken request with slade_id: NaN being sent.